doc/api/tls.md

```diff
@@ -726,14 +726,12 @@ On the client, the `session` can be provided to the `session` option of
 
 See [Session Resumption][] for more information.
 
-Note: For TLSv1.2 and below, [`tls.TLSSocket.getSession()`][] can be called once
-the handshake is complete.  For TLSv1.3, only ticket based resumption is allowed
-by the protocol, multiple tickets are sent, and the tickets aren't sent until
-later, after the handshake completes, so it is necessary to wait for the
-`'session'` event to get a resumable session.  Applications are
-recommended to use the `'session'` event instead of `getSession()` to ensure
-they will work for all TLS protocol versions.  Applications that only expect to
-get or use 1 session should listen for this event only once:
+For TLSv1.3, it is necessary to wait for the 'session' event to get a resumable
+session. This is because only ticket-based resumption is allowed by the
+protocol. Tickets aren't sent until after the handshake completes. Applications
```

Comment:
I think this doesn't quite explain to a frustrated user why they have to use this new-fangled "session" event, and can't just call getSession() once the handshake is established. The reason for that is the tickets aren't available yet, multiple tickets are (or can be, openssl defaults to 2, but its technically configurable, though I didn't expose that knob into node) sent, and the timing of them isn't synced with the TLS1.3 handshake completion anymore, they are just going to get sent "sometime after the handshake" -- and maybe not at all if the server just calls `.end()` after handshake establishment as our unit tests often do :-(... though real code would probably exchange at least a little bit of data after going through all the trouble of making a tls connection. Just some background, if you think its too much detail, leave it  out.

Reply:
@sam-github I've restored much of the removed text, greatly reducing the extent of the changes here.